### PR TITLE
Droid scorekeeper scorecard

### PR DESCRIPTION
**Pull request by Saadnajmi**
Facebook Authentication and Scorekeeping working

Only need to add the REST endpoint so that it actually posts scores

### CODE REVIEW

**Comment by Saadnajmi** [comment]
The Common.Common.Score model has HomePoints and AwayPoints as doubles. Shouldn't they be int's? We only support incrementing and decrememnting the score by integers, and I've been using them as integers this entire time.

**Comment by zuhairp** [comment]
It was mostly a just in case thing

**Comment by Saadnajmi** [comment]
you're still awake? :o

**Comment by zuhairp** [comment]
With #98 you also get event information for free, you can use that in the games list to make it more clear which competition they are scorekeeping for. #73 might give you an inspiration of how that list could possibly look like

File: mobile/Scorekeeper/Scorekeeper.Droid/Resources/values/secrets.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
```
Comment (zuhairp) [inline mobile/Scorekeeper/Scorekeeper.Droid/Resources/values/secrets.xml]: Rename this file to ```secrets.xml.example```

File: mobile/Scorekeeper/Scorekeeper.Droid/Screens/GameListActivity.cs [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+namespace Scorekeeper.Droid
+{
+    [Activity(Label = "List of Games")]
+    public class GameListActivity : Activity, IGameListView
+    {
+        string IGameListView.AccessToken
+        {
+            get
+            {
+                return AccessToken.CurrentAccessToken.Token;
```
Comment (zuhairp) [inline mobile/Scorekeeper/Scorekeeper.Droid/Screens/GameListActivity.cs]: Yeah, GameList should not even know about the access token

File: mobile/Scorekeeper/Scorekeeper.Droid/Screens/MainActivity.cs [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+            profileTracker = new MyProfileTracker();
+            profileTracker.mOnProfileChanged += OnProfileChanged;
+            profileTracker.StartTracking();
+
+            SetContentView(Resource.Layout.Main);
+
+            loginButton = FindViewById<LoginButton>(Resource.Id.login_button);
+            loginButton.SetReadPermissions("email");
+            loginButton.RegisterCallback(callbackManager, this);
```
Comment (zuhairp) [inline mobile/Scorekeeper/Scorekeeper.Droid/Screens/MainActivity.cs]: Forward the call from the loginButton to a LoginPresenter who makes the token exchange and writes the backend token to storage (using SharedPreferences). You can ignore the writing to storage part for this PR, but we'll need an implementation of ```IStorage``` that writes to SharedPreferences

File: mobile/Scorekeeper/Scorekeeper.Droid/Screens/MainActivity.cs [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+        void IFacebookCallback.OnError(FacebookException error)
+        {
+            throw new NotImplementedException();
+        }
+
+        void IFacebookCallback.OnSuccess(Java.Lang.Object result)
+        {
+            //throw new NotImplementedException();
+            LoginResult loginResult = result as LoginResult;
+            accessToken = loginResult.AccessToken.Token;
```
Comment (zuhairp) [inline mobile/Scorekeeper/Scorekeeper.Droid/Screens/MainActivity.cs]: Yeah, here you would send the access token to the login presenter

File: mobile/Scorekeeper/Scorekeeper.Droid/Screens/ScoreCardActivity.cs [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+                return int.Parse(awayScoreTV.Text);
+            }
+
+            set
+            {
+                awayScoreTV.Text = value.ToString();
+            }
+        }
+
+        int IScoreCardView.HomeScoreDelta
```
Comment (zuhairp) [inline mobile/Scorekeeper/Scorekeeper.Droid/Screens/ScoreCardActivity.cs]: I'll repeat myself - I don't think this is visually necessary. Barely anyone thinks about the delta score, and instead focuses on what the total score is. You'll need the delta for internal tracking purposes but it just adds noise to what should be a simple app to use

File: mobile/Scorekeeper/Scorekeeper.Droid/Screens/ScoreCardActivity.cs [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+            {
+                postScoreButton.Enabled = value;
+            }
+        }
+
+        void IScoreCardView.EndGame()
+        {
+            postScoreButton.Enabled = false;
+            postScoreButton.Visibility = Android.Views.ViewStates.Gone;
+            endGameButton.Text = "Game Over";
```
Comment (zuhairp) [inline mobile/Scorekeeper/Scorekeeper.Droid/Screens/ScoreCardActivity.cs]: Give a confirmation dialog first. Let them know that only an LSC admin can resume the game

File: mobile/Scorekeeper/Scorekeeper.Droid/Screens/ScoreCardActivity.cs [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+            homeScoreDelta = FindViewById<TextView>(Resource.Id.home_score_delta_TextView);
+            awayScoreDelta = FindViewById<TextView>(Resource.Id.away_score_delta_TextView);
+
+            homePlusOneButton = FindViewById<Button>(Resource.Id.home_score_plus_one_button);
+            awayPlusOneButton = FindViewById<Button>(Resource.Id.away_score_plus_one_button);
+            HomeMinusOneButton = FindViewById<Button>(Resource.Id.home_score_minus_one_button);
+            AwayMinusOneButton = FindViewById<Button>(Resource.Id.away_score_minus_one_button);
+
+
+            homePlusOneButton.Click += (sender, e) => { presenter.UpdateScore(ScoreCardPresenter.Team.Home, 1); };
```
Comment (zuhairp) [inline mobile/Scorekeeper/Scorekeeper.Droid/Screens/ScoreCardActivity.cs]: Actually, do keep the Team enum :)

File: mobile/Scorekeeper/Scorekeeper.Droid/Screens/ScoreCardActivity.cs [hunk 8]
```diff
@@ -0,0 +1,194 @@
...
+
+        protected override void OnPause()
+        {
+            base.OnPause();
+            presenter.RemoveView();
+        }
+
+        private Dialog createAlertDialog()
+        {
+            //set alert for End game button
```
Comment (zuhairp) [inline mobile/Scorekeeper/Scorekeeper.Droid/Screens/ScoreCardActivity.cs]: Oh, didn't see this

File: mobile/Common/Common.Common/API/AdminRESTClient.cs [hunk 9]
```diff
@@ -136,7 +142,13 @@ public AdminRESTClient()
 			return Game.FromJSON(jsonString);
 		}
 
-		public async Task DeleteGame(Game g)
+        public async Task EndGame(ClientModel.Game g)
+        {
+            var path = $"/admin/scorekeeper/games/{g.ID}/in-progress";
+            await PutAsync(path, "{ \"in_progress\": false}");
```
Comment (Saadnajmi) [inline mobile/Common/Common.Common/API/AdminRESTClient.cs]: This works so I don't need to?

Comment (zuhairp) [inline mobile/Common/Common.Common/API/AdminRESTClient.cs]: ok, that's fine

File: mobile/Scorekeeper/Scorekeeper.Droid/Screens/ScoreCardActivity.cs (hunk 6 above)
Comment (Saadnajmi) [inline mobile/Scorekeeper/Scorekeeper.Droid/Screens/ScoreCardActivity.cs]: There is a dialog. The code for that is in the activity though

**Pull request merged**